### DISCUSSION TITLE
Copilot completions showing, but not filling on remote Pycharm

### DISCUSSION THREAD

**Discussion by loodvn** [community/community#27961] [Copilot Conversations]
Hi there,

I'm using copilot with PyCharm plugin (1.1.28.1744).

My copilot suggestions are showing successfully, but then can't be applied when I press tab.
I'm using a remote PyCharm development environment, and according to the debugging logs this is the error:
`2022-08-01 16:58:50,734 [ 66842] WARN - #c.j.r.a.BackendActionExecutorDelegate - Action copilot.applyInlays (class com.github.copilot.actions.CopilotApplyInlaysAction) is disabled and won't be executed`

Is it possible that the Copilot plugin is enabling suggestions, but disabling applying completions? 
Completions work fine on my local PyCharm installation (also PyCharm 2022.2 and copilot v1.1.28.1744).

The only difference is that locally I see a "Logout from GitHub" option on my Copilot plugin, and on the remote IDE it doesn't show that, which makes me believe that copilot might have disabled it. 

However, I can't find a way to login to github again through the copilot plugin (to authenticate), so I assume it's still using the same working credentials as my local version...

```
System details:

My machine: MacOS 12.4, MacBook Pro M1
IDE version on my machine: "JetBrains Client 2022.2
Build #JBC-222.3345.131, built on July 27, 2022

Subscription is active until February 1, 2023.

For educational use only.

Runtime version: 17.0.3+7-b469.32 aarch64

VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.

Controller in Remote Development (PY)"

Server: CentOS Linux release 7.9.2009 (Core)

IDE version on server:
PyCharm 2022.2
{

"appPid": 1755,

"appVersion": "PY-222.3345.131",

"runtimeVersion": "17.0.3b469.32",

"unattendedMode": true,

"backendUnresponsive": false,

"modalDialogIsOpened": false,

"idePath": "/home/lov701/.cache/JetBrains/RemoteDev/dist/85174e1c13db8_pycharm-professional-2022.2",
```

**Comment by loodvn** [comment]
Relevant JetBrains YouTrack issue: https://youtrack.jetbrains.com/issue/CWM-6886/Possible-remote-specific-error-Github-Copilot-suggestions-showing-but-not-completing

**Comment by haoliuj-hub** [comment]
Have The solution? I use JetBrains Client also have the problem.

**Comment by udkl** [comment]
I do not see suggestions at all when using JetBrains Client with Webstorm. This is the first time I used the copilot plugin.

```
JetBrains Client 2022.2.2
Build #JBC-222.4167.31, built on September 14, 2022

Runtime version: 17.0.4+7-b469.53 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o

Controller in Remote Development (WS)

Remote is Ubuntu 22

```

**Comment by n0rthwood** [comment]
anyupdate on this ? it seems this is still a problem as of 2023-03-09

**Comment by ChrisCrossCrash** [comment]
Copilot seems to be working with the Jetbrains Client after the latest updates (Copilot plugin version 1.2.5.2507)! It didn't work until after I removed the Copilot extension from from the host, so the extension was installed on the **client only**.

I'm using "JetBrains Client 2023.1 Beta" on Windows 11 as the client (launched through PyCharm on WIndows) and PyCharm Ubuntu for WSL 2 as the host. I think this is the PyCharm on Ubuntu for WSL 2 is PyCharm 2023.1 (Professional Edition) Build #PY-231.8109.197. I'm using the Copilot plugin version 1.2.5.2507 on the client only.

**Comment by LucioMansilla** [comment]
the same using "code with me" extension, only the host can accept the suggestion...

**Comment by aless673** [comment]
Still same issue (PhpStorm). Really bad experience for developers...
Tried VS Code but.. really I miss my PhpStorm

**Comment by j0eii** [comment]
Experienced the same issue

**Comment by helderlj** [comment]
This is still going.... PHPStorm + WSL + Copilot Plugin = frustration. It works flawlessly on VSCode.

**Comment by bearbite** [comment]
Hi all! Here is the solution to the problem: https://youtrack.jetbrains.com/issue/GTW-3086/Possible-remote-specific-error-Github-Copilot-suggestions-showing-but-not-completing

"- add the Github Coopilot plugin ONLY FOR THE CLIENT (Settings/Plugins client) and not for the host! Remove or disable the Coopilot plugin from the host!
- Close the GoLand (or any Jetbrains IDE) and restart the remote server"

**Comment by Trolann** [comment]
Latest IDE/Remote Development updates from Jetbrains appear to correct the issue. Installing Copilot as a host plugin on WSL2 or Ubuntu machines are working with tab as expected now; tested in PyCharm and CLion 

@loodvn you might be able to close

**Comment by LucioMansilla** [comment]
still not working on Code With Me application.. 
the host is the only who can accept the suggest of copilot.

**Comment by gillbates** [comment]
same here.
workaround:
1, uninstall copilot from plugin host
2, install copilot in plugin client.

**Reply by DyadicGit** [reply]
helped, 
thought disabling on Host would be enough, but uninstalling solved the issue & Copilot started working as expected

**Comment by bgrieder** [comment]
If the Copilot plugin is installed on both the host and the client, they must be both uninstalled before re-installing the one on the client  only (this is tested on RustRover)

**Reply by aidandeno** [reply]
This is the only set of steps that resolved this issue for me

**Reply by sebavuye** [reply]
Thank you. This was indeed the only way to make it work.

**Comment by hieudmg** [comment]
Is there anyway that make it work with the plugin installed on both client and host? I work on my company's computer directly and remotely, turning the plugin on and off daily is not ideal for me.

**Comment by Prismoven** [comment]
still a problem now
stand with hieudmg
